s3/pkg/helper/functional.go

```diff
@@ -0,0 +1,19 @@
+package helper
+
+func Filter(xs []string, f func(string) bool) []string {
+	ans := make([]string, 0, len(xs))
+	for _, x := range xs {
+		if f(x) {
+			ans = append(ans, x)
+		}
+	}
+	return ans
+}
+
+func Map(xs []string, f func(string) string) []string {
+	ans := make([]string, 0, len(xs))
+	for _, x := range xs {
```

Comment:
sorry, DNU this code...what exactly is this supposed to be doing ? (it is missing an IF check ?)